src/consensus/validation.h

```diff
@@ -37,24 +37,21 @@ class CValidationState {
     std::string strDebugMessage;
 public:
     CValidationState() : mode(MODE_VALID), nDoS(0), chRejectCode(0), corruptionPossible(false) {}
-    bool DoS(int level, bool ret = false,
-             unsigned int chRejectCodeIn=0, const std::string &strRejectReasonIn="",
-             bool corruptionIn=false,
-             const std::string &strDebugMessageIn="") {
+    bool Invalid(int level, unsigned int chRejectCodeIn, const std::string& strRejectReasonIn, const std::string& strDebugMessageIn = "")
+    {
         chRejectCode = chRejectCodeIn;
         strRejectReason = strRejectReasonIn;
-        corruptionPossible = corruptionIn;
         strDebugMessage = strDebugMessageIn;
         if (mode == MODE_ERROR)
```

Comment:
Probably scope creep so feel free to ignore, but since you're changing things around so much, it'd be clearer to me to change this to:

```
if (mode != MODE_ERROR)
```

and unify the `return false`s at the end of the function.